### DISCUSSION TITLE
Problem adding trait to block

### DISCUSSION THREAD

**Discussion by JuniorVezintana** [GrapesJS/grapesjs#5258] [General]
I have a problem, when trying to add a Trait, it does not appear when selecting the item. Do the traits have a plugin to install or are they already integrated?
$this.objBlocksDefault['h1'] = {
  label: 'Headline',
  category: $this.strBlockText,
  media: `<img src="/builder/svg/h1.svg" class='icon-tools' style="width:45px">`,
  resizable: true,
  attributes: { class: '', title: "Adicione uma Headline" },
  content:
    `   
      <h1 class="" data-bold="inherit" data-gramm="false" style="display:flex; justify-content:center; align-items:center; padding: 0; margin: 5px 5px 10px 5px; text-align: center; box-sizing: border-box;  font-size: 39px;width:100%;" data-gjs-resizable="true">
        Headline 01
      </h1>
      `,
  traits: {
    content: {
      type: 'text',
      label: 'Content',
      default: 'Headline 01',
    },
  },
};

**Comment by artf** [comment]
You can't add traits on the Block level, you have to define them on the [Component](https://grapesjs.com/docs/modules/Components.html) level, then you can use that component in your block

**Reply by JuniorVezintana** [reply]
And how would I do that? do you have any example?

**Reply by artf** [reply]
It's all documented, please read about [Components](https://grapesjs.com/docs/modules/Components.html), [Traits](https://grapesjs.com/docs/modules/Traits.html) and [Blocks](https://grapesjs.com/docs/modules/Blocks.html)

**Reply by JuniorVezintana** [reply]
I managed to add the trait, when selecting the component it appears in the designated area, but when changing its content it does not update:
$this.editor.DomComponents.addType('h1', {
            isComponent: el => el.tagName == 'H1',
            model: {
              defaults: {
                traits: [
                  {
                    type: 'h1',
                    label: 'Texto do título',
                    name: 'texto',
                  },
                  {
                    type: 'color',
                    label: 'Cor do título',
                    name: 'corDoTitulo',
                    default: {
                      value: '#FF0000',
                    },
                  },
                ],
              },
            },
          });

**Reply by artf** [reply]
what is the code for the trait of `type: 'h1'`?
